odoo/addons/base/models/res_partner.py

```diff
@@ -638,6 +638,12 @@ def _company_dependent_commercial_sync(self):
                 ('company_id', '!=', self.env.company.id),
             ])
             for prop in parent_properties:
+                # unlink existing property for current partner
+                self.env['ir.property'].search([
+                    ('fields_id', 'in', prop.fields_id.ids),
```

Comment:
```suggestion
                    ('fields_id', 'in', company_dependent_commercial_field_ids),
```
and move it outside the loop to unlink the values for all the company dependent fields at once ?

Reply:
Just to be sure, we want to unlink all company-dependent properties linked to the partner and aren't part of the current company?

Reply:
Exactly